src/main/java/frc/robot/subsystems/LedSubsystem.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import java.util.Map;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.PWM;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.shuffleboard.SimpleWidget;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.commands.LedToggleCommand;
+
+public class LedSubsystem extends SubsystemBase {
+  private ShuffleboardTab ledTab = Shuffleboard.getTab("Led");
+  private GenericEntry ledState = ledTab.add("OnOff", LedToggleCommand.LedToggle).getEntry();
+  private SimpleWidget ledMode = ledTab.add("Color", LedGPMode).withWidget(BuiltInWidgets.kBooleanBox).withProperties(Map.of("colorWhenFalse", "Purple", "colorWhenTrue", "Yellow"));
+  private PWM pwm;
+  public static boolean LedGPMode = true; //True = Cone, False = Cube
+  private final AddressableLEDBuffer buffer;
+  // TODO Led parameters and RIO ports
+  private static final int length = 1; //Amount of Leds in strip light. Or one if we are using a single light
+  private static final int port = 0;
+  public final static AddressableLED leds = new AddressableLED(port);
+  public static enum LedMode {
+    CUBE, CONE
+  }
+  /** Creates a new Leds. */
+  public LedSubsystem() {
+    LedDriver(1); //Setchannel
+    buffer = new AddressableLEDBuffer(length);
+    leds.setLength(length);
+    leds.setData(buffer);
+  }
+  /**Updates mode*/    
+  public void update() {
+    LedMode ledmode;
+      // Select LED mode
+      if (LedGPMode) {
+        ledmode = LedMode.CONE;
+      } else {
+        ledmode = LedMode.CUBE;
+      }
+      setMode(ledmode);
+    }
+  /** Sends to Led blinkin */
+  public void setMode(LedMode mode){
+      switch (mode) {
+          case CUBE: 
+            solid(Color.kPurple);
+          break;
+          case CONE:
+            solid(Color.kYellow);
+          break;
+      }
+      leds.setData(buffer);
+      pwm.setSpeed(-0.99);
+  }
+  /* Toggles Mode */
+  private void solid(Color color) {
+    for (int i = 0; i < length; i++) {
+        buffer.setLED(i, color);
+    }
+  }
+  /* Makes Pwm Channel */
+  public void LedDriver(int channel) {
+    pwm = new PWM(channel);
+    pwm.setBounds(1,1, 1, 1, 1); /*TODO Setbounds */
+    pwm.setPeriodMultiplier(PWM.PeriodMultiplier.k1X);
+  }
+  @Override
+  public void periodic() {
+    update();
+    ledState.setBoolean(LedToggleCommand.LedToggle);
+    ledMode.getEntry().setBoolean(LedGPMode);
+  }
+
+/* If we are using a light strip
```

Comment:
Delete this commented section if we're not using it, we can use git to recover any experimental code we might need.